arcs/src/algorithms/scale_non_uniform.rs

```diff
@@ -0,0 +1,101 @@
+use crate::{
+    primitives::{Point, Line},
+    Vector,
+};
+use kurbo::Affine;
+
+/// Something which can be scaled **non-uniform** in x and y directions in *Drawing Space*
+pub trait ScaleNonUniform {
+    fn scale_nu(&mut self, factor_x: f64, factor_y: f64, base: Vector);
```

Comment:
Instead of `_nu`, what about using the full `_non_uniform` to improve readability?

Otherwise here are some synonyms for *uniform*:

- constant
- consistent
- steady
- invariable
- unvarying
- unfluctuating
- unvaried
- unchanging
- unwavering
- undeviating
- stable
- static
- sustained
- regular
- fixed
- even
- equal
- equable
- monotonous
- identical
- matching
- similar
- same
- alike
- like
- selfsame
- homogeneous (`scale_non_homogeneous()` sounds fairly accurate, but it's pretty long...)
- corresponding
- equivalent

Reply:
reading through this list i lean towards non-**constant**, although *non-uniform* comes from another CAD package i frequently use.

Definitely agree on the readability.

Reply:
Could the word "constant" be confused with Rust's concept of mutability?

I'm leaning towards `scale_non_uniform()`. The thinking being that trait scales something, but the scaling doesn't happen uniformly in all directions.

Reply:
One could argue that it would be `scale_mutable` then 😉 
I'm totally fine with `scale_non_uniform`, lets name it that.